lib/spack/spack/environment/environment.py

```diff
@@ -541,8 +541,12 @@ def regenerate(self, concretized_specs):
             raise SpackEnvironmentViewError(msg)
         rename(tmp_symlink_name, self.root)
 
-        # remove old_root
-        if old_root and os.path.exists(old_root):
+        # remove old_root when it looks like an old root
+        if (
+            old_root and
+            os.path.exists(old_root) and
+            os.path.samefile(os.path.dirname(new_root), os.path.dirname(old_root))
```

Comment:
Maybe a comment here:
```suggestion
            # Make sure the old root doesn't link to a file not managed by the environment
            os.path.samefile(os.path.dirname(new_root), os.path.dirname(old_root))
```
would make the edge case more explicit?

Reply:
Added that